src/main/java/org/apache/commons/csv/CSVParser.java

```diff
@@ -765,7 +765,13 @@ CSVRecord nextRecord() throws IOException {
         final long startCharPosition = lexer.getCharacterPosition() + this.characterOffset;
         do {
             this.reusableToken.reset();
-            this.lexer.nextToken(this.reusableToken);
+            try {
+                this.lexer.nextToken(this.reusableToken);
+            } catch (IOException ioe) {
+                String errorMessage = "Exception during parsing at line: "
+                        + this.lexer.getCurrentLineNumber() + ", position: " + this.lexer.getCharacterPosition();
```

Comment:
Wait a sec, why is code in the parser instead of in the lexer? All this catch does it extract information out of the lexer, so why not have exception thrown by the lexer provide this information in the first place in a better message?

Reply:
I will have a look on this...
Thanks for the review @garydgregory

Reply:
@garydgregory 
I had a look at what you are suggesting. Below are my thoughts...
If we look at the internals of method `Token nextToken(final Token token) throws IOException{}` in class `Lexer`, there are around 10 places where `IOException` can be thrown. 
Among them method calls to `parseEncapsulatedToken()`, `parseSimpleToken()` and `reader.read()` have high possibility of throwing parsing related `IOException`.

In that case we have following options.
1. handle each of method calls for customized IOException throwings with our enhanced exception message.
2. wrap `nextToken()` in class `Lexer` entirely and throw `IOException` with enhanced exception message.
3. keep the `this.lexer.nextToken(this.reusableToken)` wrapped with try catch in class `CSVParser` and rethrow IOException with enhanced message - this is what we have at the moment in PR.

IMO, keeping the try catch block in the current place we have at the moment seems more simple rather than handling exception in multiple places inside `Lexer` class.

Appreciate your opinion on this.

Reply:
I vote for number 1. try to keep exception logic in one place.